x-pack/plugins/rollup/__jest__/integration_tests/job_create.test.js

```diff
@@ -0,0 +1,1298 @@
+
+      it('should have the horizontal step active on "Terms"', () => {
+        expect(getEuiStepsHorizontalActive()).toContain('Terms');
+      });
+
+      it('should have the title set to "Terms"', () => {
+        expect(testSubjectExists('rollupJobCreateTermsTitle')).toBe(true);
+      });
+
+      it('should have a link to the documentation', () => {
+        expect(testSubjectExists('rollupJobCreateTermsDocsButton')).toBe(true);
+      });
+
+      it('should have the "next" and "back" button visible', () => {
+        expect(testSubjectExists('rollupJobBackButton')).toBe(true);
+        expect(testSubjectExists('rollupJobNextButton')).toBe(true);
+        expect(testSubjectExists('rollupJobSaveButton')).toBe(false);
+      });
+
+      it('should go to the "Date histogram" step when clicking the back button', async () => {
+        userActions.clickPreviousStep();
+        expect(getEuiStepsHorizontalActive()).toContain('Date histogram');
+      });
+
+      it('should go to the "Histogram" step when clicking the next button', async () => {
+        userActions.clickNextStep();
+        expect(getEuiStepsHorizontalActive()).toContain('Histogram');
+      });
+
+      it('should have a button to display the list of terms to chose from', () => {
+        expect(testSubjectExists('rollupJobTermsFieldChooser')).toBe(false);
+
+        findTestSubject('rollupJobShowFieldChooserButton').simulate('click');
+
+        expect(testSubjectExists('rollupJobTermsFieldChooser')).toBe(true);
+      });
+    });
+
+    describe('terms field chooser (flyout)', () => {
+      describe('layout', () => {
+        beforeEach(async () => {
+          await goToStepAndOpenFieldChooser();
+        });
+
+        it('should have the title set to "Add terms fields"', async () => {
+          expect(findTestSubject('rollupJobCreateFlyoutTitle').text()).toEqual('Add terms fields');
+        });
+
+        it('should have a button to close the flyout', () => {
+          expect(testSubjectExists('rollupJobTermsFieldChooser')).toBe(true);
+
+          findTestSubject('euiFlyoutCloseButton').simulate('click');
+
+          expect(testSubjectExists('rollupJobTermsFieldChooser')).toBe(false);
+        });
+      });
+
+      describe('when no terms are availalbe', () => {
+        it('should indicate it to the user', async () => {
+          mockIndexPatternValidityResponse({ numericFields: [], keywordFields: [] });
+          await goToStepAndOpenFieldChooser();
+
+          const { tableCellsValues } = getMetadataFromEuiTable('rollupJobTermsFieldChooser-table');
+
+          expect(tableCellsValues).toEqual([['No items found']]);
+        });
+      });
+
+      describe('when terms are available', () => {
+        beforeEach(async () => {
+          mockIndexPatternValidityResponse({ numericFields, keywordFields });
+          await goToStepAndOpenFieldChooser();
+        });
+
+        it('should display the numeric & keyword fields available', async () => {
```

Comment:
Similar to my other comments, I wonder if we could move this into tests for the specific component, i.e. `components/field_list/field_list.test.js`?

Reply:
As I commented above, this would not bring anything and to the contrary would break the point of testing a section as a whole.